x/examples/outline-cli/dns_posix.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+)
+
+const (
+	resolvConfFile           = "/etc/resolv.conf"
```

Comment:
I'm not very sure this is portable. Sometimes the file is auto generated and shouldn't be edited. Your config may get overwritten.

Reply:
That's why I'm renaming the original file to a backup file. Though [Outline Client is writing directly to the file](https://github.com/Jigsaw-Code/outline-client/blob/master/tools/outline_proxy_controller/outline_proxy_controller.cpp#L555-L590), which is not good, especially when it is a symbolic link.

Reply:
Your solution is definitely an improvement. But I think it's still problematic. Some systems may use [resolved](https://wiki.archlinux.org/title/systemd-resolved), for instance.
Let's not fix it in this PR, but we need to think about more robust solutions for DNS configuration on Linux. Perhaps we check for the presence of a few popular systems.

Reply:
Right, that's a limitation caused by Linux not providing a reliable API. We must adapt to different system configurations.